src/builders/libraryBuilder.ts

```diff
@@ -89,9 +89,16 @@ export class LibraryBuilder {
 
       logger.debug(`generating CQL definitions for profile ${structureDef.name}`);
 
-      const resourceHandler: Handler | null = createHandler(structureDef, this.valueSets);
-      if (resourceHandler !== null) {
-        resources.push(resourceHandler.process());
+      if (structureDef.type in handlerLookup) {
+        const resourceHandlerClass : typeof Handler = handlerLookup[structureDef.type];
+        const resourceHandler: Handler | null  = new resourceHandlerClass(structureDef, this.valueSets);
+
+        if (resourceHandler !== null) {
+          resources.push(resourceHandler.process());
+        }
```

Comment:
Hmm I guess this might be a redundant null check? since if structureDef.type is in the handler lookup then we are guaranteed to get an instance of the class, not null?

So I feel like in the first if block we can just do the push/process call without the second null check. I think

Reply:
Yeah that makes sense, because we defined handlerLookup to return something of type Handler, so as far as I can tell, there would be no way for resourceHandler to be null. I'll remove the `if` statement.